File: java/Wallet.java
import java.util.Scanner;

public class Wallet {
    static CategoriesManager categoriesManager = new CategoriesManager();
    static CostManager costManager = new CostManager();
    static TypeManager typeManager = new TypeManager();
    static DocsManager docsManager = new DocsManager();
    public static void prompt(){
        final int quitOption = 15;
        Scanner scanner = new Scanner(System.in);

        while (true) {
            System.out.println("\n===================================WALLET=====================================\n" +
                    "                        SUIVI DE SOLDE\n" +
                    "                                1. consulter votre solde\n" +
                    "                        GESTION DE DEPENSE\n" +
                    "                                2. Ajouter de l'argent\n" +
                    "                                3. Retirer de l'argent\n" +
                    "                        PLANIFICATION BUDGETAIRE\n" +
                    "                                4. Voir toutes les catégories\n" +
                    "                                5. Ajouter une nouvelle catégorie\n" +
                    "                                6. Supprimer une catégorie\n" +
                    "                                7. Modifier une catégorie\n" +
                    "                        GESTION DE DOCUMENTS\n" +
                    "                                8. Voir tous les types\n" +
                    "                                9. Voir tous les documents\n" +
                    "                                10. Ajouter un type de document\n" +
                    "                                11. Voir les détails sur un document\n" +
                    "                                12. Ajouter un document\n" +
                    "                                13. Retirer un document\n" +
                    "                                14. Modifier un document\n" +
                    "                               "+ quitOption + ". QUITTER\n" +
                    "================================================================================");

            System.out.print("Que souhaitez-vous faire? (réponse : un nombre de 1 à " + quitOption + ") : ");
            int option = scanner.nextInt();
            scanner.nextLine();
            switch (option) {
                case 1:
                    categoriesManager.checkBalance();
                    break;
                case 2:
                    costManager.add();
                    break;
                case 3:
                    costManager.throwBack();
                    break;
                case 4:
                    categoriesManager.displayCategories();
                    break;
                case 5:
                    categoriesManager.add();
                    break;
                case 6:
                    categoriesManager.throwBack();
                    break;
                case 7:
                    categoriesManager.modifyCategory();
                    break;
                case 8:
                    typeManager.displayTypes();
                    break;
                case 9:
                    docsManager.displayDocs();
                    break;
                case 10:
                    typeManager.add();
                    break;
                case 11:
                    docsManager.displayDocDetails();
                    break;
                case 12:
                    docsManager.add();
                    break;
                case 13:
                    docsManager.throwBack();
                    break;
                case 14:
                    docsManager.modify();
                    break;
                case quitOption:
                    return;
                default:
                    System.out.println("Veuillez choisir un nombre entre 1 et 7.");
                    break;
            }
        }
    }
}

File: python/wallet.py
from categories import display_categories, check_balance, add_money, throw_back_money, throw_back_money, add_category, throw_back_category, modify_category
from docs import add_type, display_types, display_docs, add_doc, remove_doc, display_doc_details, modify_doc

def wallet():
    quit_option = 15
    while True:
        print('''
        ===================================WALLET=====================================
                            SUIVI DE SOLDE
                                1. consulter votre solde
                            GESTION DE DEPENSE
                                2. Ajouter de l'argent
                                3. Retirer de l'argent
                            PLANIFICATION BUDGETAIRE
                                4. Voir toutes les catégories
                                5. Ajouter une nouvelle catégorie
                                6. Supprimer une catégorie
                                7. Modifier une catégorie
                            GESTION DE DOCUMENTS
                                8. Voir tous les types
                                9. Voir tous les documents
                                10. Ajouter un type de document
                                11. Voir les détails sur un document
                                12. Ajouter un document
                                13. Retirer un document
                                14. Modifier un document
                                15. QUITTER
        ===============================================================================''')

        choice = int(input(f"Que souhaitez-vous faire? (réponse : un nombre de 1 à {quit_option}) : "))

        if choice == 1:
            check_balance()
        elif choice == 2:
            add_money()
        elif choice == 3:
            throw_back_money()
        elif choice == 4:
            display_categories()
        elif choice == 5:
            add_category()
        elif choice == 6:
            remove_category()
        elif choice == 7:
            modify_category()
        elif choice == 8:
            display_types()
        elif choice == 9:
            display_docs()
        elif choice == 10:
            add_type()
        elif choice == 11:
            display_doc_details()
        elif choice == 12:
            add_document()
        elif choice == 13:
            remove_doc()
        elif choice == 14:
            modify_doc()
        elif choice == quit_option:
            return
        else:
            print(f"Veuillez choisir un nombre entre 1 et {quit_option}.")

if __name__ == "__main__":
    wallet()